### PR TITLE
OEA-463: Added a link to the trial summary page to let users permanently delete the trial

### PR DESCRIPTION
**Pull request by marshl**

### CODE REVIEW

**Comment by marshl** [comment]
I feel like the "Delete Trial" link should be red or something. Does anyone have a colour or style preference?

File: controllers/TrialController.php [hunk 1]
```diff
@@ -382,6 +386,15 @@ public function actionTransitionState()
         echo $result;
     }
 
+    public function actionDelete()
+    {
+        $trial = $this->loadModel($_POST['id']);
+        $trial->deepDelete();
```
Comment (apigram) [inline controllers/TrialController.php]: Cascading delete, eh? I like that idea!

File: models/Trial.php [hunk 2]
```diff
@@ -104,6 +104,7 @@ public function rules()
         return array(
             array('name, owner_user_id, pi_user_id, status', 'required'),
             array('name', 'length', 'max' => 64),
+            array('name', 'unique', 'caseSensitive' => false),
```
Comment (apigram) [inline models/Trial.php]: This looks like a rebase issue.

Comment (marshl) [reply models/Trial.php]: Nope, just waiting on reviews

**Comment by marshl** [comment]
@apigram @Vegemash @Zhaolian-Zhou post-rebase bump

**Comment by marshl** [comment]
@apigram @Vegemash @Zhaolian-Zhou Post-rebase bump

**Comment by marshl** [comment]
bump

**Comment by marshl** [comment]
bump

**Comment by marshl** [comment]
@apigram @Vegemash @Zhaolian-Zhou does this PR need code review, or can I just merge it?

**Comment by marshl** [comment]
@Zhaolian-Zhou GOod point, I had wondered that myself when I wrote this. But with the changes to trial statuses in the next PR, there is no longer a concept of "starting" a trial.Trials are started as soon as they begin.

File: models/Trial.php [hunk 3]
```diff
@@ -664,6 +664,55 @@ public function transitionState($new_state)
         return self::RETURN_CODE_OK;
     }
 
+    public function closeTrial()
```
Comment (marshl) [inline models/Trial.php]: Note: This isn't really meant to be here, but it is important for the next commit

Comment (apigram) [reply models/Trial.php]: Fair enough. I'll let it slide for now.

Comment (marshl) [reply models/Trial.php]: You are too kind

**Pull request merged**